### PR TITLE
Add sparse page to dense page translation support to trace_indices().

### PR DESCRIPTION
**Pull request by avose**
This adds support to trace_indices() to map the indices / addresses in the idx[] array from sparse pages that could be far apart from eachother with lage gaps into a group of consecutive pages that are dense.  The associated addresses / indices in idx[] are then re-written to reference the new, dense pages.

### CODE REVIEW

File: src/sgbuf.c [hunk 1]
```diff
@@ -99,23 +99,56 @@ size_t trace_indices( sgIdx_t *idx, size_t len, struct trace tr) {
...
+	if( !(pages = realloc(pages,npages*sizeof(long))) ) {
+	  fprintf(stderr,"trace_indices(): Failed to allocate new page entry (%ld).\n",npages);
+	}
+	pages[pidx] = page;
+      }
+      // Replace sparse page bits in address with dense page index bits.
+      new_idx  = (pidx << page_bits) | ((idx[i]*8) & ((1l<<page_bits)-1l));
+      new_idx /= 8;
+      idx[i] = new_idx;
+      if( idx[i] > new_max ) {
```
Comment (plavin) [inline src/sgbuf.c]: I believe your code should have the ability to reduce the amount of memory required, correct? In that case, should we check to see if `max` can be lowered at the end of your code?

Comment (plavin) [reply src/sgbuf.c]: Besides that, the code looks good.

Comment (avose) [reply src/sgbuf.c]: Yes, the idea to have max lowered. Here is an example:

./spatter -l 1000000 -k gather -b OpenMP -config-file ../../app_traces_stats_v2/amg.pat 
page compression: 18446744073709551582 25281151

That is, without compacting touched pages, one would need 18446744073709551582*8 bytes, but when only using touched pages and compacting them you only need 25281151*8 bytes. IDK, might be off slightly here, but I think you get the idea. Without this, the memory requirements are intractable.

Comment (avose) [reply src/sgbuf.c]: That's *8 at the end of both of those btw.

Comment (plavin) [reply src/sgbuf.c]: Sorry, I misread a bit of the code. Looks good.

**Pull request merged**